platform/engineering/design-docs/2020-05-26-content-validation.md

```diff
@@ -0,0 +1,208 @@
+
+### Testing Plan
+
+1. Configure a scheduled job to run a script once per workday
+2. Write an initial script that logs when the script is run
+3. Once the scheduled job is running at the correct time, replace the initial script with the actual script that runs the content validation
+
+### Logging
+
+Currently, invalid content is logged during staging deploys. And failed builds trigger generic [Slack notifications](https://dsva.slack.com/archives/C37M86Y8G/p1591099447374800).
+
+### Debugging
+
+##### Broken links
+
+1. Confirm that the content build is passing
+2. Modify an HTML file, like [`header.html`](https://github.com/department-of-veterans-affairs/content-build/blob/master/src/site/includes/header.html) to have a broken link
+3. Run the build again to confirm that it is failing
+
+##### Accessibility errors
+
+1. Confirm that the content build is passing
+2. Modify an HTML file, like [`header.html`](https://github.com/department-of-veterans-affairs/content-build/blob/master/src/site/includes/header.html) to have an accessibility error
+3. Run the build again to confirm that it is failing
+
+### Caveats
+
+- Our [accessibility check](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/assets/js/execute-axe-check.js#L8) only renders a banner on the preview server.
+
+### Security Concerns
+
+There are no new security concerns with a separated build process.
+
+### Privacy Concerns
+
+There are no new privacy concerns with a separated build process.
+
+### Open Questions and Risks
+
+#### Risks
+
+- Neither the current content validation nor the proposed content validation will **prevent** invalid content from making it to production. But this proposal would prevent unrelated teams from being blocked by invalid content.
+
+#### Questions
+
+### Work Estimates
+
+The following estimates vary greatly depending on who's doing the work.
+
+1. Finish extracting the `content-build` repo out of the `vets-website` repo - ?
+2. Write and test a scheduled job to run the content build - <1 day
+
+### Alternatives
+
+#### Highest quality, but lowest speed
+
+If we could proactively validate CMS content before it's published, we could eliminate a source of downstream broken links and/or accessibility errors, which would improve the experience for end users and/or reduce the number of frontend deploys that require manual intervention. The downside of this approach would be figuring out how to add the content validation to the CMS, and/or adding friction to the content writers workflow.
+
+#### Fastest speed, but lowest quality
+
+We could stop breaking the `vets-website` build due to invalid content. That would reduce the burden on the content writers and/or reduce the number of deploys requiring manual intervention. However, that would likely mean end users experience broken links and/or accessibility errors more often.
```

Comment:
This line seems like it can be interpreted as contradicting an earlier statement:

> We could stop breaking the `vets-website` build due to invalid content.

The previous statement:
> Neither the current content validation nor the proposed content validation will **prevent** invalid content from making it to production.